ngraph/src/ngraph/op/reduce_logical_and.cpp

```diff
@@ -34,3 +36,39 @@ shared_ptr<Node> op::v1::ReduceLogicalAnd::clone_with_new_inputs(const OutputVec
     check_new_args_count(this, new_args);
     return make_shared<op::v1::ReduceLogicalAnd>(new_args.at(0), new_args.at(1), get_keep_dims());
 }
+
+namespace
+{
+    void evaluate_reduce_logical_and(const HostTensorPtr& data,
+                                     const HostTensorPtr& axes,
+                                     const HostTensorPtr& out)
+    {
+        const auto axes_count = axes->get_element_count();
+        const auto axes_buffer = axes->get_data_ptr<int64_t>();
+        const AxisSet reduction_axes(
+            std::vector<AxisSet::value_type>(axes_buffer, axes_buffer + axes_count));
+
+        runtime::reference::reduce_logical_and(data->get_data_ptr<char>(),
```

Comment:
Are data types always equal to char for logical and?
Can the input data be int64 or something like that?

Reply:
Both `ReduceLogicalAnd` and `ReduceLogicalOr` inherit from the `LogicalReduction` class. Its validation method makes sure that the input data type is `boolean` https://github.com/openvinotoolkit/openvino/blob/master/ngraph/src/ngraph/op/util/logical_reduction.cpp#L113-L115

The boolean element type is 8-bits long and is usually represented by `char` https://github.com/openvinotoolkit/openvino/blob/master/ngraph/src/ngraph/type/element_type.cpp#L78 and in fact this is how it was already done in constant folding passes for those ops prior to my changes.